Contribute/Code-samples/index.md

```diff
@@ -0,0 +1,131 @@
+
+## Good practices
+
+In order to ensure the quality and consistency of all the various code samples in the Umbraco Documentation, we have outlined a set of good practices that we highly recommend that you follow, when adding code snippets to articles.
+
+* Add a clear description
+* Use code comments
+* Use real-life samples
+* Add correct syntax highlighting
+* Add only compilable samples
+* Check for syntax errors
+
+Each of these guidelines is explained in more detail below.
+
+### Add a clear description
+
+Having a code sample in an article without explaining what the code does and how it can be used, could leave the reader confused and potentially they could end up running into issues when using the snippet.
+
+Make sure to **always** add a clear description of what the code sample showcases before or after adding the snippet to the article. It should be clear where and when the snippet can be used.
+
+### Use code comments
+
+When you are adding code samples that contain more than a single feature or a single method, it is highly recommended that you add inline-comments directly in the code.
+
+By adding comments directly in the code you can avoid having too much text surrounding the code sample, and you also help readers understand each section or part of the code snippet in detail.
+
+Note that the use of code comments does not eliminate the need for a description of the code sample in the surrounding text.
+
+### Use real-life samples
+
+The documentation often aims to explain complex scenarios and concepts within Umbraco, which means that code samples can be very useful to further the understanding. It is important that these code samples are *real-life* clear examples. For example, using variables such as 'foo' and 'bar' can distract from the intent of the example. Aim to use an example that would make sense to add to a production environment.
+
+It can be a good idea to use *placeholders* for names, methods and other things, in order to keep the code samples as neutral and general as possible, althought that is not a requirement.
+
+With Umbraco, depending on the context of the implementation and the skillset of the development team - often there is more than one way to achieve a result. Having multiple examples, eg a Modelsbuilder version, a non-Modelsbuilder version etc - can help prevent readers from mixing techniques in their solution, so try to consider the context of alternative approaches, and its fine to provide multiple examples.
+
+### Add correct syntax highlighting
+
+When you add code blocks to an Umbraco Docs article make sure that you add the correct syntax highlighting. This will "prettify" the code in the sample, based on which language is used. In the Umbraco Documentation most commonly used languages are supported for syntax highlighting.
```

Comment:
May want to inform about the "supported" syntax highlighting languages?
The package we use - Skybrud.SyntaxHighlighter - doesn't support everything, they have a list of supported languages here: https://github.com/abjerner/Skybrud.SyntaxHighlighter/blob/master/src/Skybrud.SyntaxHighlighter/Language.cs

Reply:
Fx. No SQL highlighting 🙂

Reply:
Yeah, good point. I'll add the list to the Docs, with a link to the package as well 💪

Reply:
Article has been updated with the suggestions!